Assets/Scripts/Features/Gameplay/Take Photo/GeneratePhoto.cs

```diff
@@ -10,28 +8,41 @@ public class GeneratePhoto : MonoBehaviour
     [SerializeField] private GameObject _picture;
     [SerializeField] private RenderTexture _renderTexture;
 
-    void Start() {
+    /// <summary>
+    /// Execute once when the component has been created.
+    /// </summary>
+    private void Start()
+    {
         PhotoCameraApi.OnCreatePhoto += GenerateNewPhoto;
     }
 
     /// <summary>
     /// Create a new photograph.
     /// </summary>
-    void GenerateNewPhoto() {
-       Renderer rend = _picture.GetComponent<Renderer>();
+    private void GenerateNewPhoto()
+    {
+        Renderer rend = _picture.GetComponent<Renderer>();
         rend.material.mainTexture = ToTexture2D(_renderTexture);
     }
 
     /// <summary>
-    /// Returns a Texture2D generated from RenderTexture "rTex".
+    /// Cleanup when destroying the component.
+    /// </summary>
+    private void OnDestroy()
+    {
+        // Remove the reference so that OnCreatePhoto won't call the function on a destroyed instance of this script.
+        PhotoCameraApi.OnCreatePhoto -= GenerateNewPhoto;
+    }
+
+    /// <summary>
+    /// Returns a Texture2D generated from a given RenderTexture.
     /// </summary>
-    /// <param name="rTex"></param>
-    /// <returns></returns>
-    Texture2D ToTexture2D(RenderTexture rTex)
+    /// <param name="renderTexture"></param>
+    private Texture2D ToTexture2D(RenderTexture renderTexture)
```

Comment:
The parameter name is more descriptive, but add a description on the param like this: `<param name="renderTexture">renderTexture is used for ... </param>`
Also describe the return value: `<returns>modified texture</returns>`
[https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/returns](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/returns)